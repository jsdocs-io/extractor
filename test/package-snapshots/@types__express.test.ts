import { describe, it } from "vitest";
import { expectPackageAPIToMatchSnapshot } from "../helpers/expect-package-api-to-match-snapshot";

describe("@types/express", () => {
  const name = "@types/express";

  it("4.17.11", async () => {
    await expectPackageAPIToMatchSnapshot({ name, version: "4.17.11" });
  });
});
