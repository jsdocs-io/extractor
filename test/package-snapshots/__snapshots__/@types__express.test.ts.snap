// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@types/express 4.17.11 1`] = `
{
  "declarations": {
    "classes": [],
    "enums": [],
    "functions": [
      {
        "docs": [
          "/**
 * Creates an Express application. The express() function is a top-level function exported by the express module.
 */",
        ],
        "id": "e",
        "kind": "FunctionDeclaration",
        "name": "e",
        "signature": "e: typeof e;",
        "source": {
          "filename": "index.d.ts",
          "line": 27,
          "unpkgURL": "https://unpkg.com/browse/@types/express@4.17.11/index.d.ts#L27",
          "url": "https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express/index.d.ts#L27",
        },
        "type": "typeof e",
      },
      {
        "docs": [
          "/**
     * This is a built-in middleware function in Express. It parses incoming request query parameters.
     */",
        ],
        "id": "query",
        "kind": "FunctionDeclaration",
        "name": "query",
        "signature": "query: (options: qs.IParseOptions | typeof qs.parse) => Handler;",
        "source": {
          "filename": "index.d.ts",
          "line": 69,
          "unpkgURL": "https://unpkg.com/browse/@types/express@4.17.11/index.d.ts#L69",
          "url": "https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express/index.d.ts#L69",
        },
        "type": "(options: qs.IParseOptions | typeof qs.parse) => Handler",
      },
      {
        "docs": [],
        "id": "Router",
        "kind": "FunctionDeclaration",
        "name": "Router",
        "signature": "Router: (options?: RouterOptions) => core.Router;",
        "source": {
          "filename": "index.d.ts",
          "line": 71,
          "unpkgURL": "https://unpkg.com/browse/@types/express@4.17.11/index.d.ts#L71",
          "url": "https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express/index.d.ts#L71",
        },
        "type": "(options?: RouterOptions) => core.Router",
      },
    ],
    "interfaces": [
      {
        "docs": [],
        "id": "Application",
        "kind": "InterfaceDeclaration",
        "members": {
          "callSignatures": [],
          "constructSignatures": [],
          "indexSignatures": [],
          "methods": [],
          "properties": [],
        },
        "name": "Application",
        "signature": "interface Application extends core.Application {}",
        "source": {
          "filename": "index.d.ts",
          "line": 94,
          "unpkgURL": "https://unpkg.com/browse/@types/express@4.17.11/index.d.ts#L94",
          "url": "https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express/index.d.ts#L94",
        },
      },
      {
        "docs": [],
        "id": "CookieOptions",
        "kind": "InterfaceDeclaration",
        "members": {
          "callSignatures": [],
          "constructSignatures": [],
          "indexSignatures": [],
          "methods": [],
          "properties": [],
        },
        "name": "CookieOptions",
        "signature": "interface CookieOptions extends core.CookieOptions {}",
        "source": {
          "filename": "index.d.ts",
          "line": 95,
          "unpkgURL": "https://unpkg.com/browse/@types/express@4.17.11/index.d.ts#L95",
          "url": "https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express/index.d.ts#L95",
        },
      },
      {
        "docs": [],
        "id": "Errback",
        "kind": "InterfaceDeclaration",
        "members": {
          "callSignatures": [],
          "constructSignatures": [],
          "indexSignatures": [],
          "methods": [],
          "properties": [],
        },
        "name": "Errback",
        "signature": "interface Errback extends core.Errback {}",
        "source": {
          "filename": "index.d.ts",
          "line": 96,
          "unpkgURL": "https://unpkg.com/browse/@types/express@4.17.11/index.d.ts#L96",
          "url": "https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express/index.d.ts#L96",
        },
      },
      {
        "docs": [],
        "id": "ErrorRequestHandler",
        "kind": "InterfaceDeclaration",
        "members": {
          "callSignatures": [],
          "constructSignatures": [],
          "indexSignatures": [],
          "methods": [],
          "properties": [],
        },
        "name": "ErrorRequestHandler",
        "signature": "interface ErrorRequestHandler<
    P = core.ParamsDictionary,
    ResBody = any,
    ReqBody = any,
    ReqQuery = core.Query,
    Locals extends Record<string, any> = Record<string, any>
> extends core.ErrorRequestHandler<P, ResBody, ReqBody, ReqQuery, Locals> {}",
        "source": {
          "filename": "index.d.ts",
          "line": 97,
          "unpkgURL": "https://unpkg.com/browse/@types/express@4.17.11/index.d.ts#L97",
          "url": "https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express/index.d.ts#L97",
        },
      },
      {
        "docs": [],
        "id": "Express",
        "kind": "InterfaceDeclaration",
        "members": {
          "callSignatures": [],
          "constructSignatures": [],
          "indexSignatures": [],
          "methods": [],
          "properties": [],
        },
        "name": "Express",
        "signature": "interface Express extends core.Express {}",
        "source": {
          "filename": "index.d.ts",
          "line": 104,
          "unpkgURL": "https://unpkg.com/browse/@types/express@4.17.11/index.d.ts#L104",
          "url": "https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express/index.d.ts#L104",
        },
      },
      {
        "docs": [],
        "id": "Handler",
        "kind": "InterfaceDeclaration",
        "members": {
          "callSignatures": [],
          "constructSignatures": [],
          "indexSignatures": [],
          "methods": [],
          "properties": [],
        },
        "name": "Handler",
        "signature": "interface Handler extends core.Handler {}",
        "source": {
          "filename": "index.d.ts",
          "line": 105,
          "unpkgURL": "https://unpkg.com/browse/@types/express@4.17.11/index.d.ts#L105",
          "url": "https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express/index.d.ts#L105",
        },
      },
      {
        "docs": [],
        "id": "IRoute",
        "kind": "InterfaceDeclaration",
        "members": {
          "callSignatures": [],
          "constructSignatures": [],
          "indexSignatures": [],
          "methods": [],
          "properties": [],
        },
        "name": "IRoute",
        "signature": "interface IRoute extends core.IRoute {}",
        "source": {
          "filename": "index.d.ts",
          "line": 106,
          "unpkgURL": "https://unpkg.com/browse/@types/express@4.17.11/index.d.ts#L106",
          "url": "https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express/index.d.ts#L106",
        },
      },
      {
        "docs": [],
        "id": "IRouter",
        "kind": "InterfaceDeclaration",
        "members": {
          "callSignatures": [],
          "constructSignatures": [],
          "indexSignatures": [],
          "methods": [],
          "properties": [],
        },
        "name": "IRouter",
        "signature": "interface IRouter extends core.IRouter {}",
        "source": {
          "filename": "index.d.ts",
          "line": 107,
          "unpkgURL": "https://unpkg.com/browse/@types/express@4.17.11/index.d.ts#L107",
          "url": "https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express/index.d.ts#L107",
        },
      },
      {
        "docs": [],
        "id": "IRouterHandler",
        "kind": "InterfaceDeclaration",
        "members": {
          "callSignatures": [],
          "constructSignatures": [],
          "indexSignatures": [],
          "methods": [],
          "properties": [],
        },
        "name": "IRouterHandler",
        "signature": "interface IRouterHandler<T> extends core.IRouterHandler<T> {}",
        "source": {
          "filename": "index.d.ts",
          "line": 108,
          "unpkgURL": "https://unpkg.com/browse/@types/express@4.17.11/index.d.ts#L108",
          "url": "https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express/index.d.ts#L108",
        },
      },
      {
        "docs": [],
        "id": "IRouterMatcher",
        "kind": "InterfaceDeclaration",
        "members": {
          "callSignatures": [],
          "constructSignatures": [],
          "indexSignatures": [],
          "methods": [],
          "properties": [],
        },
        "name": "IRouterMatcher",
        "signature": "interface IRouterMatcher<T> extends core.IRouterMatcher<T> {}",
        "source": {
          "filename": "index.d.ts",
          "line": 109,
          "unpkgURL": "https://unpkg.com/browse/@types/express@4.17.11/index.d.ts#L109",
          "url": "https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express/index.d.ts#L109",
        },
      },
      {
        "docs": [],
        "id": "MediaType",
        "kind": "InterfaceDeclaration",
        "members": {
          "callSignatures": [],
          "constructSignatures": [],
          "indexSignatures": [],
          "methods": [],
          "properties": [],
        },
        "name": "MediaType",
        "signature": "interface MediaType extends core.MediaType {}",
        "source": {
          "filename": "index.d.ts",
          "line": 110,
          "unpkgURL": "https://unpkg.com/browse/@types/express@4.17.11/index.d.ts#L110",
          "url": "https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express/index.d.ts#L110",
        },
      },
      {
        "docs": [],
        "id": "NextFunction",
        "kind": "InterfaceDeclaration",
        "members": {
          "callSignatures": [],
          "constructSignatures": [],
          "indexSignatures": [],
          "methods": [],
          "properties": [],
        },
        "name": "NextFunction",
        "signature": "interface NextFunction extends core.NextFunction {}",
        "source": {
          "filename": "index.d.ts",
          "line": 111,
          "unpkgURL": "https://unpkg.com/browse/@types/express@4.17.11/index.d.ts#L111",
          "url": "https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express/index.d.ts#L111",
        },
      },
      {
        "docs": [],
        "id": "Request",
        "kind": "InterfaceDeclaration",
        "members": {
          "callSignatures": [],
          "constructSignatures": [],
          "indexSignatures": [],
          "methods": [],
          "properties": [],
        },
        "name": "Request",
        "signature": "interface Request<
    P = core.ParamsDictionary,
    ResBody = any,
    ReqBody = any,
    ReqQuery = core.Query,
    Locals extends Record<string, any> = Record<string, any>
> extends core.Request<P, ResBody, ReqBody, ReqQuery, Locals> {}",
        "source": {
          "filename": "index.d.ts",
          "line": 112,
          "unpkgURL": "https://unpkg.com/browse/@types/express@4.17.11/index.d.ts#L112",
          "url": "https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express/index.d.ts#L112",
        },
      },
      {
        "docs": [],
        "id": "RequestHandler",
        "kind": "InterfaceDeclaration",
        "members": {
          "callSignatures": [],
          "constructSignatures": [],
          "indexSignatures": [],
          "methods": [],
          "properties": [],
        },
        "name": "RequestHandler",
        "signature": "interface RequestHandler<
    P = core.ParamsDictionary,
    ResBody = any,
    ReqBody = any,
    ReqQuery = core.Query,
    Locals extends Record<string, any> = Record<string, any>
> extends core.RequestHandler<P, ResBody, ReqBody, ReqQuery, Locals> {}",
        "source": {
          "filename": "index.d.ts",
          "line": 119,
          "unpkgURL": "https://unpkg.com/browse/@types/express@4.17.11/index.d.ts#L119",
          "url": "https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express/index.d.ts#L119",
        },
      },
      {
        "docs": [],
        "id": "RequestParamHandler",
        "kind": "InterfaceDeclaration",
        "members": {
          "callSignatures": [],
          "constructSignatures": [],
          "indexSignatures": [],
          "methods": [],
          "properties": [],
        },
        "name": "RequestParamHandler",
        "signature": "interface RequestParamHandler extends core.RequestParamHandler {}",
        "source": {
          "filename": "index.d.ts",
          "line": 126,
          "unpkgURL": "https://unpkg.com/browse/@types/express@4.17.11/index.d.ts#L126",
          "url": "https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express/index.d.ts#L126",
        },
      },
      {
        "docs": [],
        "id": "Response",
        "kind": "InterfaceDeclaration",
        "members": {
          "callSignatures": [],
          "constructSignatures": [],
          "indexSignatures": [],
          "methods": [],
          "properties": [],
        },
        "name": "Response",
        "signature": "interface Response<
    ResBody = any,
    Locals extends Record<string, any> = Record<string, any>
> extends core.Response<ResBody, Locals> {}",
        "source": {
          "filename": "index.d.ts",
          "line": 127,
          "unpkgURL": "https://unpkg.com/browse/@types/express@4.17.11/index.d.ts#L127",
          "url": "https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express/index.d.ts#L127",
        },
      },
      {
        "docs": [],
        "id": "Router",
        "kind": "InterfaceDeclaration",
        "members": {
          "callSignatures": [],
          "constructSignatures": [],
          "indexSignatures": [],
          "methods": [],
          "properties": [],
        },
        "name": "Router",
        "signature": "interface Router extends core.Router {}",
        "source": {
          "filename": "index.d.ts",
          "line": 129,
          "unpkgURL": "https://unpkg.com/browse/@types/express@4.17.11/index.d.ts#L129",
          "url": "https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express/index.d.ts#L129",
        },
      },
      {
        "docs": [],
        "id": "RouterOptions",
        "kind": "InterfaceDeclaration",
        "members": {
          "callSignatures": [],
          "constructSignatures": [],
          "indexSignatures": [],
          "methods": [],
          "properties": [
            {
              "docs": [
                "/**
         * Enable case sensitivity.
         */",
              ],
              "id": "RouterOptions.caseSensitive",
              "isOptional": true,
              "isReadonly": false,
              "kind": "InterfacePropertyDeclaration",
              "name": "caseSensitive",
              "signature": "caseSensitive?: boolean;",
              "source": {
                "filename": "index.d.ts",
                "line": 77,
                "unpkgURL": "https://unpkg.com/browse/@types/express@4.17.11/index.d.ts#L77",
                "url": "https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express/index.d.ts#L77",
              },
              "type": "boolean",
            },
            {
              "docs": [
                "/**
         * Preserve the req.params values from the parent router.
         * If the parent and the child have conflicting param names, the child’s value take precedence.
         *
         * @default false
         * @since 4.5.0
         */",
              ],
              "id": "RouterOptions.mergeParams",
              "isOptional": true,
              "isReadonly": false,
              "kind": "InterfacePropertyDeclaration",
              "name": "mergeParams",
              "signature": "mergeParams?: boolean;",
              "source": {
                "filename": "index.d.ts",
                "line": 86,
                "unpkgURL": "https://unpkg.com/browse/@types/express@4.17.11/index.d.ts#L86",
                "url": "https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express/index.d.ts#L86",
              },
              "type": "boolean",
            },
            {
              "docs": [
                "/**
         * Enable strict routing.
         */",
              ],
              "id": "RouterOptions.strict",
              "isOptional": true,
              "isReadonly": false,
              "kind": "InterfacePropertyDeclaration",
              "name": "strict",
              "signature": "strict?: boolean;",
              "source": {
                "filename": "index.d.ts",
                "line": 91,
                "unpkgURL": "https://unpkg.com/browse/@types/express@4.17.11/index.d.ts#L91",
                "url": "https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express/index.d.ts#L91",
              },
              "type": "boolean",
            },
          ],
        },
        "name": "RouterOptions",
        "signature": "interface RouterOptions {}",
        "source": {
          "filename": "index.d.ts",
          "line": 73,
          "unpkgURL": "https://unpkg.com/browse/@types/express@4.17.11/index.d.ts#L73",
          "url": "https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express/index.d.ts#L73",
        },
      },
      {
        "docs": [],
        "id": "Send",
        "kind": "InterfaceDeclaration",
        "members": {
          "callSignatures": [],
          "constructSignatures": [],
          "indexSignatures": [],
          "methods": [],
          "properties": [],
        },
        "name": "Send",
        "signature": "interface Send extends core.Send {}",
        "source": {
          "filename": "index.d.ts",
          "line": 130,
          "unpkgURL": "https://unpkg.com/browse/@types/express@4.17.11/index.d.ts#L130",
          "url": "https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express/index.d.ts#L130",
        },
      },
    ],
    "namespaces": [],
    "typeAliases": [],
    "variables": [
      {
        "docs": [
          "/**
     * These are the exposed prototypes.
     */",
        ],
        "id": "application",
        "kind": "VariableDeclaration",
        "name": "application",
        "signature": "var application: Application;",
        "source": {
          "filename": "index.d.ts",
          "line": 51,
          "unpkgURL": "https://unpkg.com/browse/@types/express@4.17.11/index.d.ts#L51",
          "url": "https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express/index.d.ts#L51",
        },
        "type": "Application",
        "variableKind": "var",
      },
      {
        "docs": [
          "/**
     * This is a built-in middleware function in Express. It parses incoming requests with JSON payloads and is based on body-parser.
     * @since 4.16.0
     */",
        ],
        "id": "json",
        "kind": "VariableDeclaration",
        "name": "json",
        "signature": "var json: any;",
        "source": {
          "filename": "index.d.ts",
          "line": 34,
          "unpkgURL": "https://unpkg.com/browse/@types/express@4.17.11/index.d.ts#L34",
          "url": "https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express/index.d.ts#L34",
        },
        "type": "any",
        "variableKind": "var",
      },
      {
        "docs": [
          "/**
     * This is a built-in middleware function in Express. It parses incoming requests with Buffer payloads and is based on body-parser.
     * @since 4.17.0
     */",
        ],
        "id": "raw",
        "kind": "VariableDeclaration",
        "name": "raw",
        "signature": "var raw: any;",
        "source": {
          "filename": "index.d.ts",
          "line": 40,
          "unpkgURL": "https://unpkg.com/browse/@types/express@4.17.11/index.d.ts#L40",
          "url": "https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express/index.d.ts#L40",
        },
        "type": "any",
        "variableKind": "var",
      },
      {
        "docs": [],
        "id": "request",
        "kind": "VariableDeclaration",
        "name": "request",
        "signature": "var request: Request<
    core.ParamsDictionary,
    any,
    any,
    core.Query,
    Record<string, any>
>;",
        "source": {
          "filename": "index.d.ts",
          "line": 52,
          "unpkgURL": "https://unpkg.com/browse/@types/express@4.17.11/index.d.ts#L52",
          "url": "https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express/index.d.ts#L52",
        },
        "type": "Request<core.ParamsDictionary, any, any, core.Query, Record<string, any>>",
        "variableKind": "var",
      },
      {
        "docs": [],
        "id": "response",
        "kind": "VariableDeclaration",
        "name": "response",
        "signature": "var response: Response<any, Record<string, any>>;",
        "source": {
          "filename": "index.d.ts",
          "line": 53,
          "unpkgURL": "https://unpkg.com/browse/@types/express@4.17.11/index.d.ts#L53",
          "url": "https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express/index.d.ts#L53",
        },
        "type": "Response<any, Record<string, any>>",
        "variableKind": "var",
      },
      {
        "docs": [
          "/**
     * This is a built-in middleware function in Express. It serves static files and is based on serve-static.
     */",
        ],
        "id": "static",
        "kind": "VariableDeclaration",
        "name": "static",
        "signature": "var static: serveStatic.RequestHandlerConstructor<
    Response<any, Record<string, any>>
>;",
        "source": {
          "filename": "index.d.ts",
          "line": 58,
          "unpkgURL": "https://unpkg.com/browse/@types/express@4.17.11/index.d.ts#L58",
          "url": "https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express/index.d.ts#L58",
        },
        "type": "serveStatic.RequestHandlerConstructor<Response<any, Record<string, any>>>",
        "variableKind": "var",
      },
      {
        "docs": [
          "/**
     * This is a built-in middleware function in Express. It parses incoming requests with text payloads and is based on body-parser.
     * @since 4.17.0
     */",
        ],
        "id": "text",
        "kind": "VariableDeclaration",
        "name": "text",
        "signature": "var text: any;",
        "source": {
          "filename": "index.d.ts",
          "line": 46,
          "unpkgURL": "https://unpkg.com/browse/@types/express@4.17.11/index.d.ts#L46",
          "url": "https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express/index.d.ts#L46",
        },
        "type": "any",
        "variableKind": "var",
      },
      {
        "docs": [
          "/**
     * This is a built-in middleware function in Express. It parses incoming requests with urlencoded payloads and is based on body-parser.
     * @since 4.16.0
     */",
        ],
        "id": "urlencoded",
        "kind": "VariableDeclaration",
        "name": "urlencoded",
        "signature": "var urlencoded: any;",
        "source": {
          "filename": "index.d.ts",
          "line": 64,
          "unpkgURL": "https://unpkg.com/browse/@types/express@4.17.11/index.d.ts#L64",
          "url": "https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express/index.d.ts#L64",
        },
        "type": "any",
        "variableKind": "var",
      },
    ],
  },
  "files": [
    {
      "filename": "index.d.ts",
      "isIndexFile": true,
      "unpkgURL": "https://unpkg.com/browse/@types/express@4.17.11/index.d.ts",
      "url": "https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express/index.d.ts",
    },
  ],
  "overview": undefined,
}
`;
